### PR TITLE
bug fix webhook secret

### PR DESCRIPTION
**Pull request by jampulanaveen**

### CODE REVIEW

File: woo-razorpay.php [hunk 1]
```diff
@@ -513,6 +518,21 @@ public function createOrGetRazorpayOrderId($orderId, $is1ccCheckout = 'no')
             rzpLogInfo("createOrGetRazorpayOrderId $orderId and is1ccCheckout is set to $is1ccCheckout");
 
             $create = false;
+            $getWebhookFlag =  get_option('webhook_enable_flag');
```
Comment (ramth05) [inline woo-razorpay.php:521]: @jampulanaveen @ChetanGN , please verify if `get_option` works here or we need to use  `$this->getSetting`

Comment (jampulanaveen) [reply woo-razorpay.php:521]: @ramth05 webhook_enable_flag we are not saving it form settings, we are creating new field, so we could not fetch form $this->getSetting

Comment (ChetanGN) [inline woo-razorpay.php:521]: @jampulanaveen Shift this functionality to create rzp order function

File: woo-razorpay.php [hunk 2]
```diff
@@ -740,6 +732,22 @@ public function getCustomerInfo($order)
...
+            if (empty($getWebhookFlag) == false)
+            {
+                    if ($getWebhookFlag + 86400 < time())
+                    {
+                        $this->autoEnableWebhook();
+                    }
+            }
+            else
+            {
+                    update_option('webhook_enable_flag', $time);
```
Comment (ramth05) [inline woo-razorpay.php:748]: @jampulanaveen , move this down after webhook API call.

File: woo-razorpay.php [hunk 3]
```diff
@@ -307,9 +307,7 @@ public function autoEnableWebhook()
             $key_id      = $this->getSetting('key_id');
             $key_secret  = $this->getSetting('key_secret');
             $enabled     = true;
-            $alphanumericString = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ-=~!@#$%^&*()_+,./<>?;:[]{}|abcdefghijklmnopqrstuvwxyz';
-            $secret = substr(str_shuffle($alphanumericString), 0, 20);
-
+            $secret = $this->getSetting('webhook_secret') ? $this->getSetting('webhook_secret') : $this->generateSecret();
```
[the remark this answers is not in the document]
Comment (ramth05) [reply woo-razorpay.php:310]: also line spacing.

File: woo-razorpay.php (hunk 2 above)
Comment (ramth05) [reply woo-razorpay.php:748]: @jampulanaveen `while ( $webhook['count'] >= 1);` fix this one.

File: woo-razorpay.php [hunk 4]
```diff
@@ -307,8 +307,7 @@ public function autoEnableWebhook()
             $key_id      = $this->getSetting('key_id');
             $key_secret  = $this->getSetting('key_secret');
             $enabled     = true;
-            $alphanumericString = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ-=~!@#$%^&*()_+,./<>?;:[]{}|abcdefghijklmnopqrstuvwxyz';
-            $secret = substr(str_shuffle($alphanumericString), 0, 20);
+            $secret = empty($this->getSetting('webhook_secret')) ? $this->generateSecret() : $this->getSetting('webhook_secret');
```
Comment (ramth05) [inline woo-razorpay.php:310]: @jampulanaveen , `(empty($this->getSetting('webhook_secret')) === true)`

**Review by ramth05 — APPROVED** [review]
LGTM. Please apply requested change before merge and test it throughly.

**Pull request merged**